### PR TITLE
feat: support multiple cache paths

### PR DESCRIPTION
**Pull request by frezbo**
Support multiple cache paths.

### CODE REVIEW

File: internal/pkg/types/v1alpha2/steps.go [hunk 1]
```diff
@@ -28,14 +28,15 @@ func (steps Steps) Validate() error {
...
-	Env       Environment  `yaml:"env,omitempty"`
-	CachePath string       `yaml:"cache,omitempty"`
-	TmpDir    string       `yaml:"-"`
-	Sources   Sources      `yaml:"sources,omitempty"`
-	Prepare   Instructions `yaml:"prepare,omitempty"`
-	Build     Instructions `yaml:"build,omitempty"`
-	Install   Instructions `yaml:"install,omitempty"`
-	Test      Instructions `yaml:"test,omitempty"`
+	Env        Environment  `yaml:"env,omitempty"`
+	CachePath  string       `yaml:"cache,omitempty"`
```
Comment (frezbo) [inline internal/pkg/types/v1alpha2/steps.go:32]: should we remove this or just deprecate it, it's not used by any projects under siderolabs

Comment (smira) [reply internal/pkg/types/v1alpha2/steps.go:32]: was not even implemented before?

Comment (frezbo) [reply internal/pkg/types/v1alpha2/steps.go:32]: it was, but for some reason only supported a single cache path, while in theory buildkit supports multiple

**Review by smira — APPROVED** [review]

**Pull request merged**